sdk/keyvault/internal/challenge_policy.go

```diff
@@ -102,18 +99,14 @@ func (k *KeyVaultChallengePolicy) Do(req *policy.Request) (*http.Response, error
 		if token, ok := tk.(*azcore.AccessToken); ok {
 			req.Raw().Header.Set(
 				headerAuthorization,
-				fmt.Sprintf("%s %s", bearerHeader, token.Token),
+				bearerHeader+token.Token,
 			)
 		} else {
 			// tk is not an azcore.AccessToken type, something went wrong and we should return the 401 and accompanying error
 			return resp, requestErr
 		}
 
-		resp, err = http.DefaultClient.Do(req.Raw())
-		if err != nil {
-			// A second request failed, return error
-			return nil, err
-		}
+		return req.Next()
```

Comment:
This does what we want even though we called `req.Next()` already?

Reply:
Looking at the source it will throw an error, I believe we can use `http.DefaultClient` here, the request has gone through all the policies and this block only updates the authorization header.

Reply:
Not all policies modify request state. For example, we want this request to pass through the logging policy again.

Reply:
I'm not sure how to pass this request through the pipeline then. The policies attached to the original request will be emptied now and the pipeline is a private member of the `Connection` struct in the generated code. Should this policy keep a copy of an entire pipeline minus this challenge policy for this request?